common/src/chain/gen_block.rs

```diff
@@ -93,3 +115,23 @@ impl GenBlockId {
         }
     }
 }
+
+/// Classified generalized block reference
+#[derive(Eq, PartialEq, Clone, Debug)]
+pub enum GenBlockIdRef<'a> {
```

Comment:
Why do we need this type? Seems like an overkill and complicates things...

Reply:
[discussed] `GenBlockIdRef` and `Id<GenBlock>::classify_ref` have been removed.